modules/tinymce/src/core/main/ts/util/Quirks.ts

```diff
@@ -164,6 +166,25 @@ const Quirks = (editor: Editor): Quirks => {
     }
   };
 
+  const manageEnterInFigure = () => {
+    editor.on('keydown', (e) => {
+      if (e.keyCode === VK.ENTER) {
+        const currentNode = SugarElement.fromDom(editor.selection.getNode());
+        if (SugarNode.isTag('figure')(currentNode)) {
+          PredicateFind.descendant(currentNode, SugarNode.isTag('figcaption'))
+            .filter((e) => !NodeType.isContentEditableFalse(e.dom))
+            .bind((figcaption) => PredicateFind.descendant(figcaption, SugarNode.isText))
```

Comment:
It might not only be text node positions the first valid position might be before a `IMG` element. There is helper function that does this `CaretFinder.findFirstPositionIn` that will scan a element for the first valid caret position that exists if there is any. Then that caret position can be converted into a range with pos.toRange() and be set on the editor selection.

Reply:
changed